docs/madr/decisions/046-meshservice-multizone.md

```diff
@@ -0,0 +1,383 @@
+
+#### MeshService with dataplaneTags selector applied on global CP
+
+To achieve replicated service, we define a MeshService on global CP.
+This MeshService is then synced to all zones.
+
+```yaml
+kind: MeshService
+metadata:
+  name: redis-xzh64d
+  namespace: kuma-system
+  labels:
+    team: db-operators
+    kuma.io/mesh: default
+    kuma.io/origin: global
+spec:
+  selector:
+    dataplane:
+      tags: # it has no kuma.io/zone, so it aggregates all redis from redis-system from all zones
+        app: redis
+        k8s.kuma.io/namespace: redis-system
+status:
+  ports:
+  - port: 1234
+    protocol: tcp
+  addresses:
+  - hostname: redis.redis-system.svc.mesh.global
+    status: Available
+    origin: global-generator
+  vips:
+  - ip: 241.0.0.1
+    type: Kuma
+  zones: # computed on global. The rest of status fields are computed on the zone
+  - name: east
+  - name: west
+```
+
+The disadvantages:
+* We need to do a computation of list of zones on global cp, because we don't sync all Dataplanes across all zones.
+  This can be expensive because global has list of all data plane proxies.
+* We wouldn't have a capability of apply "multizone" service on specific zone, because we don't have all DPPs in all zones.
+
+**Hostname and VIP management**
+We can provide hostnames by selecting by `kuma.io/origin` label.
+
+```yaml
+type: HostnameGenerator
+name: global-generator
+spec:
+  targetRef:
+    kind: MeshServices
+    labels:
+      kuma.io/origin: global
+  template: {{ label "kuma.io/display-name" }}.{{ label "k8s.kuma.io/namespace" }}.svc.mesh.global
+```
+
+#### MeshMultiZoneService
+
+If we assume that replicated service is aggregation of multiple services, we can define a specific object for it.
+`MeshMultiZoneService` selects services instead of proxies. 
+
+```yaml
+kind: MeshMultiZoneService
+metadata:
+  name: redis-xzh64d
+  namespace: kuma-system
+  labels:
+    team: db-operators
+    kuma.io/mesh: default
+spec:
+  meshServiceSelector: # we select services by labels, not dataplane objects
+    matchLabels:
+      k8s.kuma.io/service-name: redis
+      k8s.kuma.io/namespace: redis-system
+status: # computed on the zone
+  ports:
+  - port: 1234
+    protocol: tcp
+  addresses:
+  - hostname: redis.redis-system.svc.mesh.global
+    status: Available
+    origin: global-generator
+  vips:
+  - ip: 242.0.0.1 # separate CIDR
+    type: Kuma
+  zones:
+  - name: east
+  - name: west
+```
+
+Considered names alternatives:
+* MeshGlobalService - we want to avoid using Global as it may indicate non-mesh specific object.
+* GlobalMeshService
+
+The advantages:
+* We don't need to compute anything on global cp. In the example above, both redis MeshServices have been synced to the zone.
+* We don't need to traverse over all data plane proxies
+* Users don't need to redefine ports
+* Policy matching easier to implement (will be covered more in policy matching MADR)
+* We can expose a capability to apply this locally on one zone.
```

Comment:
I don't understand this

Reply:
We can introduce functionality to apply MeshMultizoneService on Zone CP, not only on Global CP